hera_cal/delay_filter.py

```diff
@@ -193,58 +131,36 @@ def load_delay_filter_and_write(infilename, calfile=None, Nbls_per_load=None, sp
         for i in range(0, len(hd.bls), Nbls_per_load):
             df = DelayFilter(hd, input_cal=calfile)
             df.read(bls=hd.bls[i:i + Nbls_per_load], frequencies=freqs)
-            df.run_filter(cache_dir=cache_dir, read_cache=read_cache, write_cache=write_cache, **filter_kwargs)
+            df.run_delay_filter(cache_dir=cache_dir, read_cache=read_cache, write_cache=write_cache, **filter_kwargs)
             df.write_filtered_data(res_outfilename=res_outfilename, CLEAN_outfilename=CLEAN_outfilename,
                                    filled_outfilename=filled_outfilename, partial_write=True,
                                    clobber=clobber, add_to_history=add_to_history, Nfreqs=len(freqs), freq_array=np.asarray([freqs]))
             df.hd.data_array = None  # this forces a reload in the next loop
 
+# ----------------------------------------
+# Arg-parser for delay-filtering.
+# can be linear or clean.
+# ---------------------------------------
 
-def delay_filter_argparser():
-    '''Core Arg parser for commandline operation of hera_cal.delay_filter.'''
-    a = argparse.ArgumentParser(description="Perform delay filter of visibility data.")
-    a.add_argument("infilename", type=str, help="path to visibility data file to delay filter")
-    a.add_argument("--filetype_in", type=str, default='uvh5', help='filetype of input data files (default "uvh5")')
-    a.add_argument("--filetype_out", type=str, default='uvh5', help='filetype for output data files (default "uvh5")')
-    a.add_argument("--calfile", default=None, type=str, help="optional string path to calibration file to apply to data before delay filtering")
-    a.add_argument("--partial_load_Nbls", default=None, type=int, help="the number of baselines to load at once (default None means load full data")
-    a.add_argument("--res_outfilename", default=None, type=str, help="path for writing the filtered visibilities with flags")
-    a.add_argument("--clobber", default=False, action="store_true", help='overwrites existing file at outfile')
-    a.add_argument("--spw_range", type=int, default=None, nargs=2, help="spectral window of data to foreground filter.")
 
+def delay_filter_argparser(mode='clean'):
+    '''
+    Arg parser for commandline operation of delay filters.
+    Parameters
```

Comment:
minor doc string formatting issue here: needs proper line breaks and indentation

Reply:
I've added line break.